src/main/java/io/jenkins/plugins/opentelemetry/init/JenkinsExecutorMonitoringInitializer.java

```diff
@@ -42,30 +48,87 @@ public void postConstruct() {
+        //  * `jenkins.node` metric with the `status` attribute
+        //  * `jenkins.executor.total` metric with the `status` attribute
         final ObservableLongMeasurement availableExecutors = meter.gaugeBuilder(JENKINS_EXECUTOR_AVAILABLE).setUnit("${executors}").setDescription("Available executors").ofLongs().buildObserver();
         final ObservableLongMeasurement busyExecutors = meter.gaugeBuilder(JENKINS_EXECUTOR_BUSY).setUnit("${executors}").setDescription("Busy executors").ofLongs().buildObserver();
         final ObservableLongMeasurement idleExecutors = meter.gaugeBuilder(JENKINS_EXECUTOR_IDLE).setUnit("${executors}").setDescription("Idle executors").ofLongs().buildObserver();
         final ObservableLongMeasurement onlineExecutors = meter.gaugeBuilder(JENKINS_EXECUTOR_ONLINE).setUnit("${executors}").setDescription("Online executors").ofLongs().buildObserver();
         final ObservableLongMeasurement connectingExecutors = meter.gaugeBuilder(JENKINS_EXECUTOR_CONNECTING).setUnit("${executors}").setDescription("Connecting executors").ofLongs().buildObserver();
         final ObservableLongMeasurement definedExecutors = meter.gaugeBuilder(JENKINS_EXECUTOR_DEFINED).setUnit("${executors}").setDescription("Defined executors").ofLongs().buildObserver();
-        final ObservableLongMeasurement queueLength = meter.gaugeBuilder(JENKINS_EXECUTOR_QUEUE).setUnit("${items}").setDescription("Executors queue items").ofLongs().buildObserver();
+
         logger.log(Level.FINER, () -> "Metrics: " + availableExecutors + ", " + busyExecutors + ", " + idleExecutors + ", " + onlineExecutors + ", " + connectingExecutors + ", " + definedExecutors + ", " + queueLength);
 
         meter.batchCallback(() -> {
             logger.log(Level.FINE, () -> "Recording Jenkins controller executor pool metrics...");
-            logger.log(Level.FINER, () -> "Metrics: " + availableExecutors + ", " + busyExecutors + ", " + idleExecutors + ", " + onlineExecutors + ", " + connectingExecutors + ", " + definedExecutors + ", " + queueLength);
             Jenkins jenkins = Jenkins.get();
+
+            // TOTAL EXECUTORS
+            AtomicInteger totalExecutorsIdle = new AtomicInteger();
+            AtomicInteger totalExecutorsBusy = new AtomicInteger();
+            AtomicInteger nodeOnline = new AtomicInteger();
+            AtomicInteger nodeOffline = new AtomicInteger();
+
+            if (jenkins.getNumExecutors() > 0) {
+                nodeOnline.incrementAndGet();
+                Optional.ofNullable(jenkins.toComputer())
+                    .map(Computer::getExecutors)
+                    .ifPresent(e -> e.forEach(executor -> {
+                        if (executor.isIdle()) {
+                            totalExecutorsIdle.incrementAndGet();
+                        } else {
+                            totalExecutorsBusy.incrementAndGet();
+                        }
+                    }));
+            }
```

Comment:
This pattern has been added several times for different entities. It could be nice to delegate it to specialized classes and simplify the code.

Reply:
Agreed but do you have suggestions?
I looked at the Jenkins Prometheus plugin code, like the org.jenkinsci.plugins.prometheus.ExecutorCollector. It's true that hey don't have such a repetition but I didn't understand the logic of the code probably because it's very abstract